contracts/implementations/MultipleMinter.sol

```diff
@@ -146,4 +146,120 @@ contract MultipleMinter is
+     */
+    function mintVehicleAndSdWithDeviceDefinitionSign(
+        MintVehicleAndSdWithDdInput calldata data
+    ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+        VehicleStorage.Storage storage vs = VehicleStorage.getStorage();
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        address vehicleIdProxyAddress = vs.idProxyAddress;
+        address sdIdProxyAddress = sds.idProxyAddress;
+
+        if (
+            !INFT(IntegrationStorage.getStorage().idProxyAddress).exists(
+                data.integrationNode
+            )
+        ) revert InvalidParentNode(data.integrationNode);
+        if (
+            !INFT(ManufacturerStorage.getStorage().idProxyAddress).exists(
+                data.manufacturerNode
+            )
+        ) revert InvalidParentNode(data.manufacturerNode);
+        if (sds.deviceAddressToNodeId[data.syntheticDeviceAddr] != 0)
+            revert DeviceAlreadyRegistered(data.syntheticDeviceAddr);
+
+        bytes32 message = keccak256(
+            abi.encode(MINT_VEHICLE_SD_TYPEHASH, data.integrationNode)
+        );
+
+        if (
+            !Eip712CheckerInternal._verifySignature(
+                data.syntheticDeviceAddr,
+                message,
+                data.syntheticDeviceSig
+            )
+        ) revert InvalidSdSignature();
+
+        // ----- START Vehicle mint -----
+        uint256 newTokenIdVehicle = INFT(vehicleIdProxyAddress).safeMint(
+            data.owner
+        );
+
+        emit VehicleNodeMintedWithDeviceDefinition(
+            data.manufacturerNode,
+            newTokenIdVehicle,
+            data.owner,
+            data.deviceDefinitionId
+        );
+
+        message = keccak256(
+            abi.encode(
+                MINT_VEHICLE_WITH_DD_TYPEHASH,
+                data.manufacturerNode,
+                data.owner,
+                keccak256(bytes(data.deviceDefinitionId))
+            )
+        );
+
+        if (
+            !Eip712CheckerInternal._verifySignature(
+                data.owner,
+                message,
+                data.vehicleOwnerSig
+            )
+        ) revert InvalidOwnerSignature();
+        // ----- END Vehicle mint -----
+
+        // ----- START Synthetic Device mint and attributes -----
+        uint256 newTokenIdDevice = INFT(sdIdProxyAddress).safeMint(data.owner);
+
+        emit SyntheticDeviceNodeMinted(
+            data.integrationNode,
+            newTokenIdDevice,
+            newTokenIdVehicle,
+            data.syntheticDeviceAddr,
+            data.owner
+        );
+
+        if (data.attrInfoPairsDevice.length > 0)
+            _setInfos(newTokenIdDevice, data.attrInfoPairsDevice);
+        // ----- END Synthetic Device mint and attributes -----
+
+        // ----- Internal contract state change -----
```

Comment:
Another thing I thought of—is the DD something we want to "clear out" upon burn?

Reply:
hummm good point. On it